### DISCUSSION TITLE
Form wizard backed by Dataflows

### DISCUSSION THREAD

**Discussion by VaZark** [directus/directus#13788] [Feature Requests]
In most applications, users have to fill up forms that are inserted across multiple tables/collection. Directus provides a gorgeous interface for inserting data, however it is still tied to the DB itself.

What do you think about having a form builder (single/multipage) that is not tied any collection?

The form validation and data insertion can be delegated to a custom dataflow. Initial values for drop downs can be passed as props or mapped to the input field.

This allows multiple steps of logical validation and being able to invoke additional triggers right from the app without diving into creating an extension.

This could potentially be reused for the shareable forms that I saw on a discussion sometime ago

**Comment by rijkvanzanten** [comment]
That's a pretty cool idea on reusing both the publicly shared idea and data flows :)  Wondering if could make the form builder the setting for a new "Form Trigger" or something in data flows, which would effectively render out a regular html form for you to inject on a page (or maybe in a more agnostic format, so you're not locked into making websites)

**Reply by VaZark** [reply]
There could be two ways to go about it then. The “Form Trigger” could essentially be webhook that expects form-data. 

1. This will allow devs to just invoke as another API if they have an existing frontend. 
2. Directus can have its builtin opinionated form builder UI that uses this in the background. Thus the shareable form can just follow the project-level styling and will be less of a hassle managing custom styling

**Reply by benhaynes** [reply]
I really like this idea. A lot to figure out to get this right, but there's good potential with this capability. Combining this with sharable forms would be huge for crowd-sourced data collection for interconnected data.

**Comment by rijkvanzanten** [comment]
Heya! Thanks for opening this feature request!

This feature request has received over **15 votes** from the community. This means we'll move this feature request to the Under Review state! 

The Core team will schedule a meeting to review this request as soon as possible. The discussion will then be approved or denied. You may or may not be invited to join this meeting with the core team.

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._

**Reply by averri** [reply]
Hi @rijkvanzanten, thank you for the previous information. I believe this is one of the most useful feature requests, along with another one, requesting support for database views. Both features can be combined to give more flexibility in the composition of the UIs.
